onnxruntime/core/graph/graph.cc

```diff
@@ -51,13 +51,34 @@ static bool GraphLoadedFromModelFile(const GraphProto* graph_proto) {
                          graph_proto->value_info_size() != 0);
 }
 
-NodeArg::NodeArg(const std::string& name,
-                 const TypeProto* p_node_arg_type) {
+// there are some known invalid usages of dim_param and dim_value. remove them from the TypeProto so that
+// they don't affect shape inferencing or the allocation planner
+static void RemoveInvalidValues(ONNX_NAMESPACE::TypeProto& type) {
+  if (type.has_tensor_type() && type.tensor_type().has_shape()) {
+    auto* shape = type.mutable_tensor_type()->mutable_shape();
+    for (int i = 0, end = shape->dim_size(); i < end; ++i) {
+      auto& dim = *shape->mutable_dim(i);
+      if (dim.has_dim_param()) {
+        auto dim_param = dim.dim_param();
+        if (dim_param.empty() || dim_param == "None" || dim_param == "-1") {
```

Comment:
>dim_param == "None" [](start = 33, length = 19)

why special case on this string? I think it's valid from ONNX spec to have any string for dim_param.

Reply:
Some ONNX exporters apparently use string like "None". The key problem is that these exporters convert every unknown dimension to the same string "None" (even though the different unknown dimensions may not have the same value at runtime) … this is inconsistent with the ONNX spec. This change is intended to help run those models … otherwise, those models would fail with a runtime error (in cases where there is a dimension mismatch).